main.js

```diff
@@ -1,64 +1,95 @@
-var page = require('webpage').create(),
-    system = require('system'),
-    fs = require('fs'),
-    zip = new require('node-zip')(),
-    _ = require('lodash-node');
+const puppeteer = require("puppeteer");
+const fs = require("fs/promises");
+const _ = require("lodash");
+const zip = new require("node-zip")();
 
-if (system.args.length != 3) {
-  console.warn("Must provide lat,lon and output directory arguments");
-  phantom.exit();
-}
-
-var outputFile = system.args[system.args.length - 1],
-    center = system.args[system.args.length - 2];
+const lon = Number.parseFloat(process.argv[2]);
+const lat = Number.parseFloat(process.argv[3]);
+const sessionID = process.argv[4];
+const cookieKey = "_strava4_session";
 
-try {
-  fs.write(outputFile, '');
-}
-catch (e) {
-  console.warn('Cannot write to file ' + outputFile);
-  phantom.exit();
+if (!(lon && lat && sessionID)) {
+	console.log("Must include latitude, longitude and session-id");
+	console.log(
+		"Remember to use the '--' end of options flag to allow negative numbers"
+	);
+	process.exit(1);
 }
 
-if (center.split(',').length != 2) {
-  console.warn('Must supply lat,lon in the form of lat,lon')
-  phantom.exit();
-}
-else {
-  var latlon = center.split(','),
-      lat = latlon[0],
-      lon = latlon[1];
-}
+//create template file
+const createTemplate = async () => {
+	const templateData = await fs.readFile("kml.template", "utf8");
+	const kmlTemplate = _.template(templateData);
+	return kmlTemplate;
+};
+
+//execution
+(async () => {
+	const browser = await puppeteer.launch();
+	const page = await browser.newPage();
 
-var zoom = 15,
-    size = 1000,
-    east, west, north, south, bounds;
+	//disable geolocation
+	const context = browser.defaultBrowserContext();
+	await context.overridePermissions(
+		`https://www.strava.com/heatmap#14.33/${lon}/${lat}/hot/ride`,
+		["geolocation"]
+	);
 
-var kmlTemplate = _.template(fs.read('kml.template', 'utf8'));
+	page.viewport({ width: 1000, height: 1000 });
+	await page.setCookie({
+		name: cookieKey,
+		value: sessionID,
+		domain: ".strava.com",
+	});
+	await page.goto(
+		`https://www.strava.com/heatmap#14.33/${lon}/${lat}/hot/ride`,
```

Comment:
Can we go back to the gray color scheme?

Reply:
Yes